### PR TITLE
ACS-4799: Elasticsearch query feature expansion -

### PR DESCRIPTION
**Pull request by krdabrowski**
https://alfresco.atlassian.net/browse/ACS-4799

### CODE REVIEW

**Review by mpichura — APPROVED** [review]
LGTM, subject to green build.

**Review by MateuszHY — APPROVED** [review]

**Review by azakrzewski-hy — APPROVED** [review]

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    ServerHealth serverHealth;
+
+    @Autowired
+    DataUser dataUser;
+
+    @Autowired
+    DataContent dataContent;
+
+    @Autowired
+    SearchQueryService searchQueryService;
```
Comment (dsibilio) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java:46]: Can't these fields be `private`?

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    {
+        Map<String, String> templates = Map.of("_NODE", "%cm:name");
+        String query = "TYPE:'cm:content' AND ~_NODE:" + SEARCH_TERM;
+        SearchRequest request = req("afts", query, templates);
+
+        searchQueryService.expectNodeRefsFromQuery(request, testUser, fileWithTermInName.getNodeRef());
+    }
+
+    /* ********************* Helper methods ********************* */
+
```
Comment (dsibilio) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java:181]:
```suggestion

```
I think it's clear enough that anything that is not annotated with `@Test` is most likely an helper method, I'd avoid this kind of marker comments in the code as they're easy to miss and end up with Tests below it.

**Review by dsibilio — CHANGES_REQUESTED** [review]
Can't these fields be `private`?

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java (hunk 1 above)
Comment (krdabrowski) [reply tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java:46]: They should be private, somehow I've missed it 🤔 Thanks!

**Review by dsibilio — APPROVED** [review]

**Pull request merged**